README


This archive contains the distribution Dist-Zilla-Plugin-Test-CheckDeps,
version 0.012:

  Check for presence of dependencies

This software is copyright (c) 2011 by Leon Timmermans.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Dist-Zilla-Plugin-Test-CheckDeps

0.012     2013-12-20 01:21:28 +0000
          - bump dependency on Test::CheckDeps to keep pace with the version
            used by the generated test (thanks SineSwiper!)
          - bump dependency on Dist::Zilla to ensure that META.* files are
            spec-compliant, to avoid spurious test failures when combined with
            newer CPAN::Meta::Validator

0.011     2013-12-14 19:19:33 +0000
          - update configure_requires checking in Makefile.PL, add CONTRIBUTING file

0.010     2013-10-12 22:14:57 +0000
          - new "todo_when" option, for neutering tests under specific
            (runtime) conditions

0.009     2013-09-26 03:35:32 +0000
          - new 'filename' option, for changing the location of the created
            test
          - re-release with fixed compile test

0.008     2013-09-19 01:56:35 +0000
          - bumped required version of Test::CheckDeps to latest
          - added header to test, indicating the version of the plugin that
            generated it

0.007     2013-06-23 21:56:19 +0200
          - allow testing of additional phases/types as TODO tests
            (Karen Etheridge)
          - Require latest Test::CheckDeps

0.006     2013-04-28 08:39:02 -0700
          - require version of Test::More that added is_passing()
            (RT#84905, Karen Etheridge)

0.005     2012-05-02 23:40:44 +0200
          - Increase runtime dependency on Test::CheckDeps to 0.002

0.004     2012-05-02 21:54:33 +0200
          - Inject a test-phase requirement on Test::CheckDeps and Test::More
            into the distribution being built.

0.003     2012-02-06 23:39:53 +0200
          - Rename test to t/00-check-deps.t
          - Add runtime dependency on Test::CheckDeps and Test::More

0.002     2011-10-17 22:56:07 +0200
          - Renamed to Dist::Zilla::Plugin::Test::CheckDeps

0.001     2011-10-14 20:55:11 +0200
          - Initial release


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-Test-CheckDeps was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.005.
# Don't edit it but the dist.ini used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '6.30',
    'Module::Build::Tiny' => '0.030',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Check for presence of dependencies",
  "AUTHOR" => "Leon Timmermans <leont\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build::Tiny" => "0.030"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Test-CheckDeps",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::Plugin::Test::CheckDeps",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::InlineFiles" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "namespace::autoclean" => 0
  },
  "TEST_REQUIRES" => {
    "Cwd" => 0,
    "Dist::Zilla" => "4.300039",
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Path::Tiny" => 0,
    "Test::CheckDeps" => "0.010",
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Deep::JSON" => 0,
    "Test::More" => "0.94",
    "if" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.012",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



